### PR TITLE
Some refators

### PR DESCRIPTION
**Pull request by rogercloud**
1. Refactor the msg broadcast logic
2. Make kv tests parallel runnable

### CODE REVIEW

File: xline/src/server/xline_server.rs [hunk 1]
```diff
@@ -28,8 +30,6 @@ use crate::storage::KvStore;
 pub struct XlineServer {
     /// Server name
     name: String,
-    /// Address of server
-    addr: SocketAddr,
```
Comment (pzheng1025) [inline xline/src/server/xline_server.rs:32]: Ok to remove for now. May need to store this information in the future.

**Review by pzheng1025 — APPROVED** [review]

File: curp/src/sync_manager.rs [hunk 2]
```diff
@@ -104,97 +108,173 @@ impl<C: Command + 'static> SyncManager<C> {
...
+                                term,
+                                index.numeric_cast(),
+                                cmds_cloned.as_ref(),
+                            )?)
+                            .await
+                    },
+                    |r| match r.into_inner().commit_response {
+                        Some(CommitResponse::Committed(_)) => 1,
+                        Some(CommitResponse::WrongTerm(_) | CommitResponse::PrevNotReady(_)) => 0,
+                        None => unreachable!("Should contain sync response"),
```
Comment (pzheng1025) [inline curp/src/sync_manager.rs:271]: Should contain commit response

File: xline/src/server/xline_server.rs (hunk 1 above)
Comment (rogercloud) [reply xline/src/server/xline_server.rs:32]: Yes. Let's add it back later if needed.

File: curp/src/sync_manager.rs (hunk 2 above)
Comment (rogercloud) [reply curp/src/sync_manager.rs:271]: unreachable is a panic. The whole process is terminated.

**Pull request merged**